Quantum-Control-Applications/ESR/T1.py

```diff
@@ -127,78 +121,56 @@
+    )
     # ver_t2: center-to-center time between the readout window to the second pulse arriving to 'ensemble'
-    ver_t2 = (
-        analog_wf["elements"]["resonator"][0]["timestamp"] + (analog_wf["elements"]["resonator"][0]["duration"] / 2)
-    ) - (analog_wf["elements"]["ensemble"][2]["timestamp"] + (analog_wf["elements"]["ensemble"][2]["duration"] / 2))
-
-    print("center to center time between 1st and 2nd pulse", ver_t1)
-    print("center to center time between readout and 2nd pulse", ver_t2)
+    ver_t2 = get_c2c_time(job, ("ensemble", 2), ("resonator", 0))
+    print(
+        f"center to center time between 2nd pulse and readout is {ver_t2} --> internal delay to add: {ver_t2 - 4 * safe_delay} ns"
+    )
 
 else:
     qm = qmm.open_qm(config)
 
     job = qm.execute(T1)  # execute QUA program
 
-    res_handle = job.result_handles
-
-    I_handle = res_handle.get("I")
-    I_handle.wait_for_values(1)
-    Q_handle = res_handle.get("Q")
-    Q_handle.wait_for_values(1)
-    echo1_handle = res_handle.get("echo1")
-    echo1_handle.wait_for_values(1)
-    echo2_handle = res_handle.get("echo2")
-    echo2_handle.wait_for_values(1)
-    iteration_handle = res_handle.get("iteration")
-    iteration_handle.wait_for_values(1)
-    next_percent = 0.1  # First time print 10%
-
-    def on_close(event):
-        event.canvas.stop_event_loop()
-        job.halt()
-
-    f = plt.figure()
-    f.canvas.mpl_connect("close_event", on_close)
-    print("Progress =", end=" ")
-
-    while res_handle.is_processing():
+    # Get results from QUA program
+    results = fetching_tool(job, data_list=["I", "Q", "iteration"], mode="live")
+
+    fig = plt.figure()
+    interrupt_on_close(fig, job)  # Interrupts the job when closing the figure
+    while job.result_handles.is_processing():
         try:
-            I = I_handle.fetch_all()
-            Q = Q_handle.fetch_all()
-            iteration = iteration_handle.fetch_all()
-            if iteration / n_avg > next_percent:
-                percent = 10 * round(iteration / n_avg * 10)  # Round to nearest 10%
-                print(f"{percent}%", end=" ")
-                next_percent = percent / 100 + 0.1  # Print every 10%
-
-            plt.plot(waits * 4, I)
-            plt.plot(waits * 4, Q)
-            plt.title(f"iteration: {iteration}")
+            # Fetch results
+            I, Q, iteration = results.fetch_all()
+            # Convert I & Q to Volts
+            I = u.demod2volts(I, readout_len)
+            Q = u.demod2volts(Q, readout_len)
+            # Display progress bar
+            progress_counter(iteration, n_avg)
+            # Plot data
+            plt.plot(wait_vec * 4, I, label="I")
+            plt.plot(wait_vec * 4, Q, label="Q")
+            plt.xlabel("Decay time [ns]")
+            plt.ylabel("Echo magnitude I & Q [V]")
+            plt.legend()
             plt.pause(0.2)
-            plt.clf()
 
-        except Exception as e:
+        except (Exception,):
```

Comment:
If you write `except Exception as e` pycharm gives a warning "too broad exception clause" because you don't specify the kind of exception you want to catch. A way of getting rid of this warning is to write `except (Exception,)`

Reply:
While I was writing the reason why I thought it was useful I realized it was not :)
Maybe @KevinAVR has some insight?